### PR TITLE
Add Terraform reference

### PR DESCRIPTION
**Pull request by jonatasbaldin**
Hey there 👋 

This PR adds a Terraform reference, with a quick starting guide and pointing to the official docs ✨ 

TODO:
- [x] Add UTM to the signup page link

### CODE REVIEW

File: docs/tools/terraform.rst [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+See the `official documentation <https://registry.terraform.io/providers/aiven/aiven/latest/docs>`_ to learn about all the possible services and resources and the `GitHub repository <https://github.com/aiven/terraform-provider-aiven>`_ for reporting issues and contributing.
+
+.. caution::
+  Recreating stateful services with Terraform will possibly delete the service and all its data before creating it again. Whenever the Terraform plan indicates that a service will be deleted or replaced, a catastrophic action is possibly about to happen.
+
+  Some properties, like project and the resource name, cannot be changed and it will trigger a resource replacement.
+
+  To avoid any issues, please set the termination_protection property to true on all production services, it will prevent Terraform to remove the service until the flag is set back to false again. While it prevents a service to be deleted, any logical databases, topics or other configurations may be removed even when this section is enabled. Be very careful!
+
+Getting Started
```
Comment (lornajane) [inline docs/tools/terraform.rst:17]: Getting started with what? Let's add a little sentence about what we're getting you started with or to do.

File: docs/tools/terraform.rst [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+Learn More
+----------
+Check out these resources to learn more about Terraform and our Provider:
+
+* `Learn Terraform <https://learn.hashicorp.com/collections/terraform/aws-get-started>`_
+* `Aiven Terraform Provider documentation <https://registry.terraform.io/providers/aiven/aiven/latest/docs>`_
+
+Contributing
+------------
+If you have any issues or would like to contribute to the tool, please join us on the `GitHub repository <https://github.com/aiven/terraform-provider-aiven>`_.
```
Comment (lornajane) [inline docs/tools/terraform.rst:91]: :+1: happy to see this included

**Review by lornajane — CHANGES_REQUESTED** [review]
Let's flesh out what's actually happening a bit, but this is a really good guide once it has a few more words

File: docs/tools/terraform.rst [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+''''''''''''
+- `Download and install Terraform <https://www.terraform.io/downloads.html>`_
+- `Sign up <https://console.aiven.io/signup?utm_source=github&utm_medium=organic&utm_campaign=devportal&utm_content=repo>`_ for Aiven if you haven't already
+- `Generate an authentication token <https://help.aiven.io/en/articles/2059201-authentication-tokens>`_ on Aiven's console or CLI
+
+Setting up the provider
+'''''''''''''''''''''''
+Create a file named `main.tf` and add the content below:
+
+.. code:: bash
```
[the remark this answers is not in the document]
Comment (jonatasbaldin) [reply docs/tools/terraform.rst:30]: In the api_token line. There are best practices to separate code and configuration, but this is the most basic to get up and running.

**Comment by jonatasbaldin** [comment]
hey, thanks for the review as always ✨ 

added some text around the issues you've commented while trying to keep it as succinct as possible

**Review by lornajane — APPROVED** [review]
This is a good addition! Minor changes added, and I'll wait for the build to build before merging.

**Pull request merged**